File: scaffold/src/main/java/top/xuqingquan/web/IndicatorController.java
package top.xuqingquan.web;

import com.tencent.smtt.sdk.WebView;
import top.xuqingquan.web.agent.BaseIndicatorSpec;

public interface IndicatorController {

    void progress(WebView v, int newProgress);

    BaseIndicatorSpec offerIndicator();

    void showIndicator();

    void setProgress(int newProgress);

    void finish();
}

File: scaffold_web/src/main/java/top/xuqingquan/web/publics/IndicatorController.kt
package top.xuqingquan.web.publics

import android.webkit.WebView
import top.xuqingquan.web.nokernel.BaseIndicatorSpec
import com.tencent.smtt.sdk.WebView as X5WebView

interface IndicatorController {

    fun progress(v: WebView, newProgress: Int)

    fun progress(v: X5WebView, newProgress: Int)

    fun offerIndicator(): BaseIndicatorSpec?

    fun showIndicator()

    fun setProgress(newProgress: Int)

    fun finish()
}
